$MirOS: ports/devel/cvs/patches/patch-lib_test-getdate_sh,v 1.1 2010/09/18 22:35:09 tg Exp $

	-rtg mircvs://src/gnu/usr.bin/cvs/lib/getdate.y
	was patched from -rFSF to also have a different
	output format; change these tests to match that

	MirBSD: also need to disable leap seconds

--- lib/test-getdate.sh.orig	Fri Apr 15 21:12:11 2005
+++ lib/test-getdate.sh	Sat Sep 18 23:48:11 2010
@@ -1,4 +1,5 @@
 #! /bin/sh
+# $MirOS$
 
 # Test that a getdate executable meets its specification.
 #
@@ -19,6 +20,11 @@
 # Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA.  */
 
 
+# as this uses POSIX behaviour and does not count leap seconds...
+if test -n "$GETDATE_LD_PRELOAD"; then
+	LD_PRELOAD=$GETDATE_LD_PRELOAD
+	export LD_PRELOAD
+fi
 
 ###
 ### Globals
@@ -184,22 +190,22 @@ TZ=UTC0; export TZ
 
 cat >getdate-expected <<EOF
 Enter date, or blank line to exit.
-	> Bad format - couldn't convert.
-	> Bad format - couldn't convert.
-	> 1972-12-05 00:00:00.000000000
-	> 1974-03-29 00:00:00.000000000
-	> 1996-05-12 13:57:45.000000000
-	> 2012-05-12 00:00:00.000000000
-	> 1996-05-12 00:00:00.000000000
-	> Bad format - couldn't convert.
-	> Bad format - couldn't convert.
-	> 1970-01-01 02:00:00.000000000
-	> Bad format - couldn't convert.
-	> 1969-12-31 22:00:00.000000000
-	> Bad format - couldn't convert.
-	> 1970-01-01 06:00:00.000000000
-	> 1997-01-12 00:00:00.000000000
-	> 
+> Bad format - couldn't convert.
+> Bad format - couldn't convert.
+>      92361600 =	1972-12-05 00:00:00.000000000
+>     133747200 =	1974-03-29 00:00:00.000000000
+>     831909465 =	1996-05-12 13:57:45.000000000
+>    1336780800 =	2012-05-12 00:00:00.000000000
+>     831859200 =	1996-05-12 00:00:00.000000000
+> Bad format - couldn't convert.
+> Bad format - couldn't convert.
+>          7200 =	1970-01-01 02:00:00.000000000
+> Bad format - couldn't convert.
+>         -7200 =	1969-12-31 22:00:00.000000000
+> Bad format - couldn't convert.
+>         21600 =	1970-01-01 06:00:00.000000000
+>     853027200 =	1997-01-12 00:00:00.000000000
+> 
 EOF
 
 ./getdate >getdate-got <<EOF
@@ -235,22 +241,22 @@ TZ=Europe/London; export TZ
 if valid_timezone $TZ; then
 	cat >getdate-expected <<EOF
 Enter date, or blank line to exit.
-	> 2005-03-01 00:00:00.000000000
-	> 2005-03-27 00:00:00.000000000
-	> 2005-03-28 01:00:00.000000000
-	> 2005-03-28 01:00:00.000000000
-	> 2005-03-29 01:00:00.000000000
-	> 2005-03-29 01:00:00.000000000
-	> 2005-03-30 01:00:00.000000000
-	> 2005-03-30 01:00:00.000000000
-	> 2005-03-31 01:00:00.000000000
-	> 2005-03-31 01:00:00.000000000
-	> 2005-04-01 01:00:00.000000000
-	> 2005-04-01 01:00:00.000000000
-	> 2005-04-10 01:00:00.000000000
-	> 2005-04-10 01:00:00.000000000
-	> 2005-04-01 00:00:00.000000000
-	> 
+>    1109635200 =	2005-03-01 00:00:00.000000000
+>    1111881600 =	2005-03-27 00:00:00.000000000
+>    1111968000 =	2005-03-28 01:00:00.000000000
+>    1111968000 =	2005-03-28 01:00:00.000000000
+>    1112054400 =	2005-03-29 01:00:00.000000000
+>    1112054400 =	2005-03-29 01:00:00.000000000
+>    1112140800 =	2005-03-30 01:00:00.000000000
+>    1112140800 =	2005-03-30 01:00:00.000000000
+>    1112227200 =	2005-03-31 01:00:00.000000000
+>    1112227200 =	2005-03-31 01:00:00.000000000
+>    1112313600 =	2005-04-01 01:00:00.000000000
+>    1112313600 =	2005-04-01 01:00:00.000000000
+>    1113091200 =	2005-04-10 01:00:00.000000000
+>    1113091200 =	2005-04-10 01:00:00.000000000
+>    1112310000 =	2005-04-01 00:00:00.000000000
+> 
 EOF
 
 	./getdate >getdate-got <<EOF
@@ -297,20 +303,20 @@ TZ=America/New_York; export TZ
 if valid_timezone $TZ; then
 	cat >getdate-expected <<EOF
 Enter date, or blank line to exit.
-	> 2005-03-01 00:00:00.000000000
-	> 2005-02-28 18:00:00.000000000
-	> 2005-04-01 00:00:00.000000000
-	> Bad format - couldn't convert.
-	> 2005-04-30 19:00:00.000000000
-	> 2005-04-30 20:00:00.000000000
-	> 2005-05-01 00:00:00.000000000
-	> 2005-04-30 20:00:00.000000000
-	> Bad format - couldn't convert.
-	> 2005-05-31 19:00:00.000000000
-	> 2005-05-31 20:00:00.000000000
-	> 2005-06-01 00:00:00.000000000
-	> 2005-05-31 20:00:00.000000000
-	> 
+>    1109653200 =	2005-03-01 00:00:00.000000000
+>    1109631600 =	2005-02-28 18:00:00.000000000
+>    1112331600 =	2005-04-01 00:00:00.000000000
+> Bad format - couldn't convert.
+>    1114902000 =	2005-04-30 19:00:00.000000000
+>    1114905600 =	2005-04-30 20:00:00.000000000
+>    1114920000 =	2005-05-01 00:00:00.000000000
+>    1114905600 =	2005-04-30 20:00:00.000000000
+> Bad format - couldn't convert.
+>    1117580400 =	2005-05-31 19:00:00.000000000
+>    1117584000 =	2005-05-31 20:00:00.000000000
+>    1117598400 =	2005-06-01 00:00:00.000000000
+>    1117584000 =	2005-05-31 20:00:00.000000000
+> 
 EOF
 
 	./getdate >getdate-got <<EOF
